version: "3.9"

services:
  web:
    build: .
    restart: on-failure
    environment:
      - DJANGO_SECRET_KEY=${DJANGO_SECRET_KEY}
      - POSTGRES_NAME=${DB_NAME}
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - DB_HOST=${DB_HOST}
    volumes:
      - .:/code
      - static_volume:/code/static
    ports: 
      - 8000:8000
    depends_on:
      - db
    networks:
      - db_network
    entrypoint: ["python", "manage.py", "runserver", "0.0.0.0:8000"]
  db:
    image: postgres:14.1
    ports:
      - 5432:5432
    volumes:
      - db_volume:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=${DB_NAME}
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
    networks:
      - db_network

networks:
  db_network:
    driver: bridge

volumes:
  db_volume:
  static_volume: